src/Post.php

```diff
@@ -75,10 +95,16 @@ private function get_query_filter_name() {
 	 */
 	public function endpoint_args() {
 		return [
+			'id' => [
+				'default' => false,
```

Comment:
empty doesn't work - especially for slug, because an empty slug is a vlid slug (the homepage). these fields are either or (not both) so we need to detect if one is missing or blank separately.

i think it makes more sense to return an error if somebody sends an invalid id, rather than returning some random data.